src/backend/base/langflow/components/notdiamond/notdiamond.py

```diff
@@ -0,0 +1,218 @@
+        ),
+    ]
+
+    outputs = [Output(display_name="Output", name="output", method="model_select")]
+
+    def model_select(self) -> dict:
+        api_key = SecretStr(self.api_key).get_secret_value() if self.api_key else None
+        input_value = self.input_value
+        system_message = self.system_message
+        messages = self._format_input(input_value, system_message)
+
+        selected_models = []
+        mapped_selected_models = []
+        for model in self.models:
+            model_name = get_model_name(model)
+
+            if model_name in ND_MODEL_MAPPING:
+                selected_models.append(model)
+                mapped_selected_models.append(ND_MODEL_MAPPING[model_name])
+
+        payload = {
+            "messages": messages,
+            "llm_providers": mapped_selected_models,
+            "hash_content": self.hash_content,
+        }
+
+        if self.tradeoff != "quality":
+            payload["tradeoff"] = self.tradeoff
+
+        if self.preference_id and self.preference_id != "":
+            payload["preference_id"] = self.preference_id
+
+        header = {
+            "Authorization": f"Bearer {api_key}",
+            "accept": "application/json",
+            "content-type": "application/json",
+        }
+
+        response = requests.post(
+            "https://api.notdiamond.ai/v2/modelRouter/modelSelect",
+            json=payload,
+            headers=header,
+            timeout=10,
+        )
+
+        result = response.json()
+
+        if "providers" not in result:
+            # No provider returned by NotDiamond API, likely failed. Fallback to first model.
+            return self._call_get_chat_result(self.models[0], input_value, system_message)
+
+        providers = result["providers"]
+
+        if len(providers) == 0:
+            # No provider returned by NotDiamond API, likely failed. Fallback to first model.
+            return self._call_get_chat_result(self.models[0], input_value, system_message)
+
+        nd_result = providers[0]
+
+        chosen_model = None
+        for nd_model, selected_model in zip(mapped_selected_models, selected_models, strict=False):
+            if nd_model["provider"] == nd_result["provider"] and nd_model["model"] == nd_result["model"]:
+                chosen_model = selected_model
+                break
+
+        if chosen_model is None:
+            # No provider returned by NotDiamond API, likely failed. Fallback to first model.
+            return self._call_get_chat_result(self.models[0], input_value, system_message)
+
+        return self._call_get_chat_result(chosen_model, input_value, system_message)
+
+    def _call_get_chat_result(self, chosen_model, input_value, system_message):
+        result = get_chat_result(
+            runnable=chosen_model,
+            input_value=input_value,
+            system_message=system_message,
+        )
+        return {"chosen_model": get_model_name(chosen_model), "text": result}
```

Comment:
You have to return a string or Message object. `result` is a string so it should work. Returning a dictionary can cause issues.
```suggestion
        return result
```